refactor(sermant-agentcore): 重构 Nacos 客户端获取 token 逻辑

wuqiu-ai opened the pull request:
- 支持低版本Nacos服务端的支持
- 移除对 HttpLoginProcessor 和 NamingHttpClientManager 的依赖
- 直接使用 HTTP POST 请求登录接口获取 token
- 优化 token 刷新逻辑，提高代码可读性和可维护性

**What type of PR is this?**


**What this PR does / why we need it?**

Describe it.

**Which issue(s) this PR fixes？**

Fixes #

**Does this PR introduce a user-facing change?**

No

<!-- Follow this checklist to help us incorporate your contribution quickly and easily: -->

## Checklist
- [ ] Make sure there is a [GitHub_issue](https://github.com/sermant-io/Sermant/issues) related with this PR before you start working on it.
- [x] Make sure you have squashed your change to one single commit.
- [x] GitHub Actions works fine in this PR.

wuqiu-ai committed:
refactor(sermant-agentcore): 重构 Nacos 客户端获取 token 逻辑

- 移除对 HttpLoginProcessor 和 NamingHttpClientManager 的依赖
- 直接使用 HTTP POST 请求登录接口获取 token
- 优化 token 刷新逻辑，提高代码可读性和可维护性

hanbingleixue commented on sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/nacos/NacosClient.java:
```diff
@@ -390,4 +405,43 @@ private String doRequest(String url, boolean returnStatusWhenHasError) throws IO
+                    .build();
+            HttpPost httpPost = new HttpPost(url);
+            httpPost.setConfig(requestConfig);
+            if (formParams != null && !formParams.isEmpty()) {
```
It is recommended to use the utility class MapUtils

hanbingleixue commented on sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/nacos/NacosClient.java:
```diff
@@ -390,4 +405,43 @@ private String doRequest(String url, boolean returnStatusWhenHasError) throws IO
+                    return EntityUtils.toString(response.getEntity());
+                }
+                LOGGER.error("Http post request for getting all nacos keys error, the message is: {}",
+                        EntityUtils.toString(response.getEntity()));
```
When the call fails, there is a possibility that response.getEntity() is null, and EntityUtils.toString(response.getEntity()) will throw an exception

hanbingleixue commented on sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/nacos/NacosClient.java:
```diff
@@ -390,4 +405,43 @@ private String doRequest(String url, boolean returnStatusWhenHasError) throws IO
+     * @return response body
+     * @throws IOException IO exception during service invocation process
+     */
+    private String doPost(String url, Map<String, String> formParams, boolean returnStatusWhenHasError)
```
In What Scenarios Is a StatusCode Return Required When an Invoking Failure Occurs?

hanbingleixue commented on sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/nacos/NacosClient.java:
```diff
@@ -352,12 +355,24 @@ private String buildUrl(String key, String group, String namespace, boolean exac
-            tokenTtl = Long.parseLong(loginIdentityContext.getParameter(KEY_TOKEN_TTL));
+            final StringBuilder requestUrl = new StringBuilder().append(HTTP_PROTOCOL);
+            requestUrl.append(properties.getProperty(PropertyKeyConst.SERVER_ADDR))
+                    .append(LOGIN_URL);
```
StringBuilder requestUrl = new StringBuilder().append(HTTP_PROTOCOL).append(properties.getProperty(PropertyKeyConst.SERVER_ADDR))
                    .append(LOGIN_URL);

hanbingleixue commented on sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/nacos/NacosClient.java:
```diff
@@ -352,12 +355,24 @@ private String buildUrl(String key, String group, String namespace, boolean exac
+                JSONObject jsonObject = JSONObject.parseObject(result);
+                lastToken = jsonObject.getString(KEY_ACCESS_TOKEN);
+                tokenTtl = jsonObject.getLong(KEY_TOKEN_TTL);
+                lastRefreshTime = System.currentTimeMillis();
```
Set the lastRefreshTime  before calling to ensure that the client token has expired when the server token expires.

lilai23 commented:
sign DCO in your commit, you can refer this https://sermant.io/zh/document/community/contributing.html

wuqiu-ai committed:
refactor(nacos): improve token refresh logic and error handling

- Move lastRefreshTime update to prevent potential exception issues
- Use MapUtils.isEmpty() for consistent null and empty checks
- Add null check for response entity in error logging
- Add missing import for MapUtils
